core/src/main/java/de/jplag/normalization/NormalizationGraph.java

```diff
@@ -0,0 +1,78 @@
+package de.jplag.normalization;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.stream.Collectors;
+
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.SimpleDirectedGraph;
+
+import de.jplag.Token;
+
+public class NormalizationGraph {
+    private SimpleDirectedGraph<Statement, Edge> graph;
+
+    /**
+     * Construct a new normalization graph from the tokens.
+     * @param tokens The tokens used to construct the normalization graph.
+     */
+    public NormalizationGraph(List<Token> tokens) {
+        graph = new NormalizationGraphConstructor(tokens).get();
```

Comment:
The pattern new(X).getX() looks like you want a static method that does this job instead of creating an object that is dropped afterwards

Reply:
Great suggestion, cleans the code up a lot 👍 I didn't do this intitially because of concerns about object orientation but it really doesn't make much sense now that I think about it

Reply:
Okay, I actually changed my mind. This is good for the normalization itself because there is no state associated with it. But for the normalization graph construction shown here, there is a ton of state necessary. Making all of it static seems really bad. I can't really think of better method, happy to take suggestions